### PR TITLE
Adding ability for DeclarationModifiers class to support readonly structs and methods

### LINKED ISSUE #65834: The `DeclarationModifiers` class does not properly handle `readonly` keyword for `struct` \s
**Version Used**: `main`

**Steps to Reproduce**:
Issue on our side: https://github.com/dotnet/arcade/issues/11850

[DeclarationModifiers.cs#L80](https://github.com/dotnet/roslyn/blob/55259597ad41ac41e0d1888fa52a92e73318b57e/src/Workspaces/Core/Portable/Editing/DeclarationModifiers.cs#L80)

The initialization of isReadOnly in `DeclarationModifiers.From` must be extended in case ~ when ISymbol is type declaration.
```C#
public static DeclarationModifiers From(ISymbol symbol)
{
/// ...
    isReadOnly: field?.IsReadOnly == true || property?.IsReadOnly == true,
/// ...
```
`DeclarationModifiers.From(...)` is used not just for fields and properties and for `SymbolKind.NamedType` with `TypeKind.Class` and `TypeKind.Struct`: [SyntaxGenerator.cs#L612](https://github.com/dotnet/roslyn/blob/55259597ad41ac41e0d1888fa52a92e73318b57e/src/Workspaces/Core/Portable/Editing/SyntaxGenerator.cs#L612)

```C#
namespace Foo
{
public readonly struct PublicReadonlyStruct { }
}
```
After roundtrip: source -> ISymbol-> SyntaxGenerator -> source:
```C#
namespace Foo
{
public struct PublicReadonlyStruct { }
}
```
**Diagnostic Id**:

**Expected Behavior**:

**Actual Behavior**:

### LINKED ISSUE #66170: `DeclarationModifiers.From` doesn't return `readonly` for methods
https://github.com/dotnet/roslyn/blob/5730ffc208e3b7d7fe284cc26f75ae4a43703005/src/Workspaces/Core/Portable/Editing/DeclarationModifiers.cs#L80

### PR DESCRIPTION
**Pull request by kinkeadian25**
Closes #65834 Closes #66170

### CODE REVIEW

**Review by Youssef1313 — APPROVED** [review]
LGTM

File: src/Workspaces/Core/Portable/Editing/DeclarationModifiers.cs [hunk 1]
```diff
@@ -73,11 +73,12 @@ IMethodSymbol or
                 var field = symbol as IFieldSymbol;
                 var property = symbol as IPropertySymbol;
                 var method = symbol as IMethodSymbol;
+                var type = symbol as INamedTypeSymbol;
 
                 return new DeclarationModifiers(
                     isStatic: symbol.IsStatic,
                     isAbstract: symbol.IsAbstract,
-                    isReadOnly: field?.IsReadOnly == true || property?.IsReadOnly == true,
+                    isReadOnly: field?.IsReadOnly == true || property?.IsReadOnly == true || type?.IsReadOnly == true || method?.IsReadOnly == true,
```
Comment (genlu) [inline src/Workspaces/Core/Portable/Editing/DeclarationModifiers.cs:81]: Could you please add a test to cover readonly method too?

Comment (kinkeadian25) [reply src/Workspaces/Core/Portable/Editing/DeclarationModifiers.cs:81]: I did have one in here. For some reason it had me push this twice, and my test was in the last commit! Sorry for the confusion

File: src/Workspaces/CSharpTest/CodeGeneration/SyntaxGeneratorTests.cs [hunk 2]
```diff
@@ -4034,6 +4034,17 @@ class C
 }");
         }
 
+        [Fact]
```
[the remark this answers is not in the document]
Comment (kinkeadian25) [reply src/Workspaces/CSharpTest/CodeGeneration/SyntaxGeneratorTests.cs:4037]: Okay, so do I just give the number from the github issue, and then the link to it? I see some of them have a devdiv work item, but I do not have access to that I believe.

[the remark this answers is not in the document]
Comment (dibarbet) [reply src/Workspaces/CSharpTest/CodeGeneration/SyntaxGeneratorTests.cs]: Yup, the issue number and a link to the issue, random example
https://github.com/dotnet/roslyn/blob/6729a11e7bd63a315929de6d28cc83d46c1a828b/src/EditorFeatures/CSharpTest/RefactoringHelpers/RefactoringHelpersTests.cs#L19

File: src/Workspaces/CSharpTest/CodeGeneration/SyntaxGeneratorTests.cs [hunk 3]
```diff
@@ -4034,6 +4034,17 @@ class C
 }");
         }
 
+        [Fact, WorkItem(66170, " https://github.com/dotnet/roslyn/issues/66170")]
```
Comment (Youssef1313) [inline src/Workspaces/CSharpTest/CodeGeneration/SyntaxGeneratorTests.cs:4037]:
```suggestion
        [Fact, WorkItem(66170, "https://github.com/dotnet/roslyn/issues/66170")]
```

File: src/Workspaces/CSharpTest/CodeGeneration/SyntaxGeneratorTests.cs [hunk 4]
```diff
@@ -4102,6 +4113,16 @@ class C
 }");
         }
 
+        [Fact, WorkItem(65834, " https://github.com/dotnet/roslyn/issues/65834")]
```
Comment (Youssef1313) [inline src/Workspaces/CSharpTest/CodeGeneration/SyntaxGeneratorTests.cs:4116]:
```suggestion
        [Fact, WorkItem(65834, "https://github.com/dotnet/roslyn/issues/65834")]
```

**Review by genlu — APPROVED** [review]
Thanks!

**Pull request merged**